Symfony/CS/Tests/AbstractIntegrationTest.php

```diff
@@ -0,0 +1,260 @@
+/*
+ * This file is part of the PHP CS utility.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Symfony\CS\Tests;
+
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\CS\ErrorsManager;
+use Symfony\CS\FileCacheManager;
+use Symfony\CS\Fixer;
+use Symfony\CS\FixerInterface;
+
+/**
+ * Integration test helper.
+ *
+ * @author SpacePossum
+ */
+abstract class AbstractIntegrationTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @dataProvider getTests
+     */
+    public function testIntegration($testFileName, $testTitle, $fixers, $input, $expected = null)
+    {
+        $this->doTestIntegration($testFileName, $testTitle, $fixers, $input, $expected);
+    }
+
+    public function getTests()
+    {
+        $fixturesDir = realpath($this->getFixturesDir());
+        $this->assertTrue(is_dir($fixturesDir), sprintf('Given fixture dir "%s" is not a directory.', $fixturesDir));
+
+        $tests = array();
+
+        foreach (new \RecursiveIteratorIterator(new \RecursiveDirectoryIterator($fixturesDir), \RecursiveIteratorIterator::LEAVES_ONLY) as $file) {
+            if (!preg_match('/\.test$/', $file)) {
+                continue;
+            }
+
+            $test = file_get_contents($file->getRealpath());
+            $fileName = $file->getFileName();
+            if (preg_match('/--TEST--[\r\n](.*?)\s--CONFIG--[\r\n](.*?)\s--INPUT--[\r\n](.*?[\r\n]*)(?:[\r\n]--EXPECT--\s(.*)|$)/s', $test, $match)) {
+                $testTitle = $match[1];
+                $config = $this->getFixersFromConfig($fileName, $match[2]);
+                $input = $match[3];
+                $expected = isset($match[4]) ? $match[4] : null;
+                $tests[] = array($fileName, $testTitle, $config, $input, $expected);
+            } else {
+                throw new InvalidArgumentException(sprintf('Test "%s" is not valid.', $fileName));
+            }
+        }
+
+        $this->assertNotEmpty($tests, sprintf('No tests found in fixtures dir "%s".', $fixturesDir));
+
+        return $tests;
+    }
+
+    /**
+     * Returns the full path to directory which contains the tests.
+     *
+     * @return string
+     */
+    abstract protected function getFixturesDir();
+
+    /**
+     * Returns the full path to directory where the tests will writes the temporary file.
+     *
+     * @return string
+     */
+    abstract protected function getTempDir();
+
+    /**
+     * @param string           $testFileName
+     * @param string           $testTitle
+     * @param FixerInterface[] $fixers
+     * @param string           $input
+     * @param string|null      $expected
+     */
+    protected function doTestIntegration($testFileName, $testTitle, $fixers, $input, $expected = null)
+    {
+        $errorsManager = new ErrorsManager();
+        $fixer = new Fixer();
+        $fixer->setErrorsManager($errorsManager);
```

Comment:
I think we need to make this PR ready to merge first ;)
Then we will think about 2.x line.